File: Dataset/Leetcode/train/38/248.java
class Solution {
    public String XXX(int n) {
        String pre = "1";
        String res = pre;
        for (int i = 2; i <= n; i++) {
            int count = 1;
            char preChar = pre.charAt(0);
            StringBuilder temp = new StringBuilder();
            for (int j = 1; j < pre.length(); j++) {
                char curChar = pre.charAt(j);
                if (preChar == curChar) {
                    count++;
                } else {    
                    temp.append(count);
                    temp.append(preChar);
                    count = 1;
                }
                preChar = curChar;
            }
            temp.append(count);
            temp.append(preChar);
            res = temp.toString();
            pre = res;
        }
        return res;
    }
}

File: Dataset/Leetcode/test/62/248.cpp
 class Solution {
public:
    int XXX(int m, int n) {
        if(m==0||n==0)
            return 0;
        vector<vector<int>> dp(m,vector<int>(n,1));
        for(int i=1;i<m;i++){
            for(int j=1;j<n;j++)
                dp[i][j] = dp[i-1][j] + dp[i][j-1];
        }
        return dp[m-1][n-1];
    }
};

